Warn on undocumented globals

tiennou opened the pull request:
(extracted from #26, based on #34)

This adds a warning on missing documentation comments on globals.

carlosmn commented:
This feels more like something you should get as part of `-Wdocumentation`. Does that not work?

tiennou commented:
AFAIK, `-Wdocumentation` only warn about missing things in a valid docblock comment, not a missing docblock comment itself (and the docs seems to [imply](https://clang.llvm.org/docs/UsersManual.html#cmdoption-wdocumentation) that).

tiennou commented:
Rebased, I had added support for undocumented struct fields, and then thought that having a third "check" command that's compiler-output compatible would make for a nice local check. So I've ended up cleaning up the interface a little, and reimplemented both as real warnings.

The "check" mode might or might not get implemented, at least I have a list with more [undocumented stuff](https://gist.github.com/tiennou/9f6768a8489ffa6ac5b6f471fbe507f8).

tiennou commented:
So now there's a `check` command. You use it like this:
```
DOCURIUM_CFLAGS="-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include" ../docurium/bin/cm check api.docurium
```
and you should get nice compiler errors.
